app/mailers/twilio_sender.rb

```diff
@@ -46,18 +119,38 @@ def self.get_phone_numbers_from_flow_execution(execution_id)
       Rails.logger.warn e.error_message
       return
     end
-
     # Get a message out of the studio execution which we can get the To/From numbers out of
     # The opt-in/out could come from an incoming message trigger OR an existing execution
     # The message pulled from an existing execution will be the first inbound message found within the execution
     message = execution&.context&.[]('trigger')&.[]('message') || execution&.context&.[]('widgets')&.values&.select do |x|
                                                                     x&.[]('inbound')
                                                                   end&.[](0)&.[]('inbound')
-    phone_number_from = message&.[]('From')
-    phone_number_to = message&.[]('To')
+    # Alternatively, the execution could be of type SINGLE_SMS where we sent a single outbound message
+    message = execution&.context&.[]('widgets')&.values&.select { |x| x&.[]('To') }&.[](0) if message.nil?
```

Comment:
This isn't very readable, though I understand the structure is coming from Twilio. Is it possible to make this more readable using something like `dig`?